src/lib/profiles/data-management/Current/SubscriptionClient.cpp

```diff
@@ -561,6 +559,14 @@ WEAVE_ERROR SubscriptionClient::SendSubscribeRequest(void)
                 // Start, fill, and close the TLV Structure that contains ResourceID, ProfileID, and InstanceID
                 err = mDataSinkCatalog->HandleToAddress(versionedTraitPath.mTraitDataHandle, writer,
                                                         versionedTraitPath.mRequestedVersionRange);
+
+                if (err == WEAVE_ERROR_INVALID_ARGUMENT)
+                {
+                    // we should not reach this code, Locate would fail before
```

Comment:
Rephrase comment as:
" Ideally, this code will not be reached as HandleToAddress() should find the entry in the catalog as, otherwise, the earlier Locate() call would have continued. However, keeping this check here for consistency and code safety".